File: src/main/java/com/vproj/demo/controller/WelcomeController.java
package com.vproj.demo.controller;

import com.vproj.demo.model.MenuItem;
import com.vproj.demo.service.WelcomeService;

import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

/**
 *
 * @author vivek krishna varma
 */

@RestController
public class WelcomeController {
    
    @Autowired
    WelcomeService welcomeService;  
     
    //-------------------Retrieve All Menu Items--------------------------------------------------------
     
    @RequestMapping(value = "/welcome", method = RequestMethod.GET, produces = APPLICATION_JSON_VALUE)
    @ResponseBody
    public ResponseEntity<List<MenuItem>> fetchAllMenuItems() {
        List<MenuItem> menuItems = welcomeService.fetchAllMenuItems();
        if(menuItems == null || menuItems.isEmpty())	{
            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
        }
        return new ResponseEntity<>(menuItems, HttpStatus.OK);
    }
}

File: src/main/resources/public/js/controller/welcome_controller.js
"use strict";

/**
 *
 * @author vivek krishna varma
 */

App.controller("WelcomeController", ["$scope","WelcomeService",	function($scope, WelcomeService) {
    $scope.menuItems = WelcomeService.query();
  }
]);

/*App.controller('WelcomeController', ['async','$scope', 'WelcomeService', function(async,$scope,WelcomeService) {
    var self       = this;
    self.menuItems = async;
}]);*/


//App.controller('WelcomeController', ['$scope', 'WelcomeService', function ($scope, WelcomeService) {
//        
//    var self          = this;
//    self.menuItem     = new WelcomeService();
//    self.menuItems    = [];
//                
//    self.fetchAllMenuItems = function () {     
//        self.menuItems = WelcomeService.query();
//    };
//    
//    self.fetchAllMenuItems();
//}]);


//
//App.controller('WelcomeController', ['$scope', 'WelcomeService', function ($scope, WelcomeService) {
//    var self          = this;
//    self.menuItem     = new WelcomeService();
//    self.menuItems    = [];
//
//
//
//
//    self.fetchAllMenuItems = function () {
//
//        WelcomeService.query(
//            function (result, responseHeaders) {
//                self.menuItems = result;
//            }, 
//            function (httpResponse) {
//                console.log('Error while fetching welcome items list');
//            }
//        );
//    };
//}]);